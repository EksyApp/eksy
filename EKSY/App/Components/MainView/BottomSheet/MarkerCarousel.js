import React, {Component} from 'react'
import {
	View,
	StyleSheet,
	Dimensions
} from 'react-native'
import Carousel from 'react-native-snap-carousel'
import MarkerCard from './MarkerCard'
import PropTypes from 'prop-types'
import {MarkersShape} from "../../../Utils/PropTypeShapes";

const Screen = {
	width: Dimensions.get('window').width,
	height: Dimensions.get('window').height - 75
}


//Renders marker carousel view inside the bottom sheet
//Carousel contains clickable marker cards (MarkerCard) 
export default class MarkerCarousel extends Component {
	constructor(props) {
		super(props)
		this.itemWidth = (Screen.width*2)/3;
		this.sliderWidth = Screen.width
	}

	handlePress(marker) {
		this.props.setMarkerSelected(marker)
		this.props.setMarkerViewVisible()
		this.props.disableGestures(true)
	}

	renderCards() {
		return this.props.markerList.map((marker, index) => {
			return <MarkerCard
					onPress={(marker) => {
						this.handlePress(marker)
					}}
					setMarkerSelected={this.props.setMarkerSelected}
					setMarkerViewVisible={this.props.setMarkerViewVisible}
					disableGestures={this.props.disableGestures}
					width={this.itemWidth}
					marker={marker}
					key={index}
			/>
		})
	}

	render() {
		return (
				<View style={styles.container}>
					<Carousel
							itemWidth={this.itemWidth}
							sliderWidth={this.sliderWidth}
							enableSnap={true}
					>
						{this.renderCards()}
					</Carousel>
				</View>
		)
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1
	}
})

MarkerCarousel.propTypes = {
	setMarkerSelected: PropTypes.func,
	setMarkerViewVisible: PropTypes.func,
	disableGestures: PropTypes.func,
	markerList: MarkersShape,
}
