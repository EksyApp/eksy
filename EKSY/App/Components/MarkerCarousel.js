import React, { Component } from 'react'
import {
  View,
  StyleSheet
} from 'react-native'
import Carousel from 'react-native-snap-carousel'
import MarkerCard from './MarkerCard'

export default class MarkerCarousel extends Component {
  constructor (props) {
    super(props)
    this.itemWidth = 130
    this.sliderWidth = 150
  }

  renderCard () {
    return this.props.markerList.map((marker, index) => {
    <MarkerCard
    marker = {marker}
    key = {index}/>
    })
  }

  render () {
    return (
      <View style={styles.container}>
        <Carousel
          itemWidth={this.itemWidth}
          sliderWidth={this.sliderWidth}
          enableSnap={false}
       />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1
  }
})
