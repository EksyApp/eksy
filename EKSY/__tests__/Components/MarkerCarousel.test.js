import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import MarkerCarousel from "../../App/Components/MainView/MarkerCarousel";

jest.mock('react-native-snap-carousel', () => {
	const React = require.requireActual('react');
	const {View} = require('react-native')
	
	class MockCarousel extends React.Component {
		
		render() {
			return (<View {...this.props} />);
		}
	}
	MockCarousel.style = {}
	return MockCarousel
})

test('MarkerCarousel renders correctly', () => {
	const tree = renderer.create(
			<MarkerCarousel markerList={[]} />
	).toJSON();
	expect(tree).toMatchSnapshot();
});