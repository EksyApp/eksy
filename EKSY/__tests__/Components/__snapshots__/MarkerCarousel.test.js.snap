// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkerCarousel renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    enableSnap={true}
    itemWidth={500}
    sliderWidth={750}
  />
</View>
`;
